### ISSUE TITLE
Publishing on Pypi

### ISSUE THREAD

**Issue by gmaze** [euroargodev/argodmqc_owc#23]
For standard users of the software, they should not need to clone the repository to be able to run it, but rather simply install it using pip or conda.
This implies 2 things:
- a proper setup.py and packaging
- a serious refactoring of the code to manage options and configuration parameters outside of the software code (to be honest, I'd like this to be my 1st contribution to the code)

This is also to let you known that when this will be ready to publish euroargodev python softwares on pypi, I created a euroargodev account at https://test.pypi.org/user/euroargodev/ and https://www.pypi.org/user/euroargodev/
(don't be fulled by the picture, this is retrieved automatically from my email but will change soon, so that it is the EA logo).

**Comment by gmaze** [comment]
While investigating #3 , I have discovered that pip does not use underscores in package names (they are discouraged in [PEP8](https://www.python.org/dev/peps/pep-0008/#package-and-module-names)).

So, even if we work with `py_owc` internally, once published, users will have to install it with `py-owc`:
```bash
pip install py-owc
```

I dislike this idea and would prefer to work everywhere with the same name.

So, new proposition, what about a simple software named:

# pyowc

I think we still do the link with `matlab_owc`, and yet a fully embrace the python way.

**Comment by gmaze** [comment]
ok, let's go everywhere (but not repo name) with **pyowc**

**Comment by edsmall-bodc** [comment]
@apswong @gmaze I understand the conflict here - the plotting section is **not** needed to create the mapped salinity, nor is it needed to calibrate the salinity of floats.

There are two crucial points to make though:

1. The plots are needed for our DMQC operators to make well informed decisions about whether or not to accept the calibration. Without the plots, no decision can be reached, and therefore the outputs are somewhat useless.

2. One of the things we have pushed for with this is to try and create a CONSISTENT and UNIFIED approach to the decision making process, and that includes documentation. Part of this plan is that plot outputs from DMQC operators should be consistent with each other in an attempt to ensure a more consistent analysis. Therefore, we should not be implying to the operators that it is up to them to how things are displayed. We also want our operators to share their possible improvements (via github issues/pull requests), and that includes the plotting routines. I worry that, if we make the plotting routine seem optional, when they make meaningful changes it won't be shared with the rest of the community.

I realise that this could make pyowc a little more clunky to install, and more challenging to maintain, but it seems to me that, if we want this consistency, then it really does need to be included.

Interested to hear what @matdon17 and @kamwal have to say here too.

**Comment by gmaze** [comment]
re-opening this issue because we didn't decided solution 1 vs 2

ps: yes @edsmall-bodc you're right, your point 2 is very good

**Comment by matdon17** [comment]
What @edsmall-bodc has said reflects my position.  I do see there being the potential for pyowc being a module in a wider ecosystem of Argo QC tools though, and could be seen as a module of that wider landscape.  Although quite what the shape that might take is one for the future.

The only thing which has just occurred to me is whether we treat the trajectory plot alone as a module - as that is the one that seems to bring the most headaches with setup and dependencies (?).  I am not convinced in either direction though.

**Comment by edsmall-bodc** [comment]
I like this idea as well - the trajectory plot is probably the least useful plot in terms of decision making (though, if I am wrong someone is free to correct me on this) and is probably the plot that causes the biggest headache. We could make the plot optional in installation, but provide some basic instructions/suggestions on how to plot trajectory.

There is also the fact that the trajectory plot is probably the plot most likely to be able to be used by other software that might have nothing to do with salinity calibration, so it would make sense to have a stand alone trajectory plotting routine that every argo tool could use..? Or we add it as a feature in argopy? If you are fetching argo data, it is likely you might want to plot the locations, perhaps?

**Comment by gmaze** [comment]
Plotting the trajectory is definitely a generic plot
We [also have one in argopy](https://github.com/euroargodev/argopy/blob/6273ddef6c22c1676aec56d9689612dbfb8ab902/argopy/plotters.py#L209) that produces a plot like this:

![](https://argopy.readthedocs.io/en/latest/_images/trajectory_sample.png)

see here: https://argopy.readthedocs.io/en/latest/visualisation.html#trajectories-from-an-index

this is clearly a very simple plot and yet that needs a lot of tuning to suit all possible floats.
I would keep it here as it is at this time. 
We'll need to talk later about common visualization tools

**Comment by edsmall-bodc** [comment]
Something I think we should consider adding at some point is something that fetches topology data from somewhere like GEBCO for the area of interest. This would mean the plot would contain very accurate and detailed bathymetry and coastlines, which is probably important in complex regions, near sandbars and islands etc.

The topology data that I am currently using isn't very precise purely because storing all that data locally would take up a huge amount of space.

**Comment by gmaze** [comment]
Yes, I also had this in argopy :https://github.com/euroargodev/argopy/blob/6273ddef6c22c1676aec56d9689612dbfb8ab902/argopy/utilities.py#L451
for the ETOPO bathy
but this is not reliable
